consult.el

```diff
@@ -576,10 +576,37 @@ Turn ARG into a list, and for each element either:
 - split it if it a string.
 - eval it if it is an expression."
   (mapcan (lambda (x)
-            (if (stringp x)
-                (split-string-and-unquote x)
-              (ensure-list (eval x 'lexical))))
-          (ensure-list arg)))
+	    (if (stringp x)
+		(split-string-and-unquote x)
+	      (ensure-list (eval x 'lexical))))
+	  (ensure-list arg)))
+
+(defun consult--build-args-with-paths (arg search-path-list)
+  "Return ARG and SEARCH-PATH-LIST joined as a flat list of split strings.
+
+Turn ARG into a list, and for each element either:
+- split it if it a string.
+- eval it if it is an expression.
+
+If SEARCH-PATH-LIST is non-nil and not equal to '(\".\"),
+splice its paths in at the end of the arg list."
+  (let ((args (copy-sequence (mapcan (lambda (x)
+				       (if (stringp x)
+					   (split-string-and-unquote x)
+					 (ensure-list (eval x 'lexical))))
+				     (ensure-list arg)))))
+    (if (and search-path-list
+	     (not (equal search-path-list '("."))))
+	(if (member "." args)
+	    (progn (unless (and search-path-list
+				(file-directory-p (car search-path-list)))
+		     (setq args (delete "-r" args)))
```

Comment:
Why remove the -r argument here? I think you can always specify -r even if you only pass a list of files. The check if the first argument is a directory is brittle and will break if the user specifies the directories/wildcards in a different order.

Reply:
You are right.  Next push will stop removing the -r.